base/genericmemory.jl

```diff
@@ -235,13 +235,17 @@ getindex(A::Memory, c::Colon) = copy(A)
 
 ## Indexing: setindex! ##
 
-function setindex!(A::Memory{T}, x, i1::Int) where {T}
-    val = x isa T ? x : convert(T,x)::T
+function _setindex!(A::Memory{T}, x::T, i1::Int) where {T}
     ref = memoryrefnew(memoryref(A), i1, @_boundscheck)
     memoryrefset!(ref, val, :not_atomic, @_boundscheck)
     return A
 end
 
+function setindex!(A::Memory{T}, x, i1::Int) where {T}
```

Comment:
```suggestion
function setindex!(A::Memory{T}, x, i1::Int) where {T}
    @_propagate_inbounds_meta
```

Reply:
I think this is addressed by #56151 already.

Reply:
Wait, no, that's a different method!